### PR TITLE
Send the token to the backend on the initial connect

### PR DESCRIPTION
**Pull request by cquiroz**
This should be the last PR on feature/auth and then we can merge it and do better testing

### CODE REVIEW

File: common/src/main/scala/explore/AppMain.scala [hunk 1]
```diff
@@ -119,10 +119,8 @@ trait AppMain extends IOApp {
       IO.fromFuture(httpCall).map(_.body)
     }
 
-    val fetchToken: IO[String] = IO("...")
-
-    val connectParameters: IO[Map[String, Json]]            =
-      fetchToken.map(token => Map("Authorization" -> s"Bearer $token".asJson))
+    def connectParameters(vault: UserVault): IO[Map[String, Json]] =
+      Map("Authorization" -> s"Bearer ${vault.token.value}".asJson).pure[IO]
```
Comment (rpiaggio) [inline common/src/main/scala/explore/AppMain.scala:123]: I think we may want to re-read the current token here from the current vault, so that we have the latest one upon reconnection. But we can investigate a way to do that in another PR.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: common/src/main/scala/explore/AppMain.scala (hunk 1 above)
Comment (cquiroz) [reply common/src/main/scala/explore/AppMain.scala:123]: Yes we need to do that, I wasn't sure how to send a new token to the connection
